TCMSApp/src/server/app.js

```diff
@@ -3,25 +3,56 @@
 
 var express = require('express');
 var app = express();
+var methodOverride = require('method-override');
+var mongoose = require('mongoose');
+var restify = require('express-restify-mongoose');
 var bodyParser = require('body-parser');
 var favicon = require('serve-favicon');
 var logger = require('morgan');
 var port = process.env.PORT || 8001;
 var four0four = require('./utils/404')();
 
+var docs = require("express-mongoose-docs");
+
 var environment = process.env.NODE_ENV;
 
+
+var router = express.Router();
+
 app.use(favicon(__dirname + '/favicon.ico'));
 app.use(bodyParser.urlencoded({extended: true}));
 app.use(bodyParser.json());
+app.use(methodOverride())
 app.use(logger('dev'));
 
-app.use('/api', require('./routes'));
+
+//Get mongoose models
+var defect =  require('./model/mschemas/defect');
+var project =  require('./model/mschemas/project');
+var run =  require('./model/mschemas/run');
+var suite =  require('./model/mschemas/suite');
+var user =  require('./model/mschemas/user');
+
 
 console.log('About to crank up node');
 console.log('PORT=' + port);
 console.log('NODE_ENV=' + environment);
 
+
+
+// Create API with 'express-restify-mongoose'
+restify.serve(router, defect);
+restify.serve(router, project);
+restify.serve(router, run);
+restify.serve(router, suite);
+restify.serve(router, user);
+
+app.use(router)
+
+//Generate API Docs
+docs(app);
```

Comment:
I'll appreciate if you help with better solution for docs.
Since this one can be generated only "...after all express middleware..." (from documentation), so I'm not sure if it is possible to move generation to other place.